packages/material-ui-lab/src/internal/pickers/PickersPopper.tsx

```diff
@@ -87,27 +86,37 @@ const PickersPopper: React.FC<PickerPopperProps & WithStyles<typeof styles>> = (
 
     if (open) {
       lastFocusedElementRef.current = document.activeElement;
+
+      // Focus the paper element when popper opens, otherwise onBlur will not be invoked.
+      // Remove "disableAutoFocus" prop on TrapFocus also works but TrapFocus seems buggy.
+      // The page will jump a little bit when the button that opens the picker is
+      // located at the bottom of the screen.
+      paperRef.current?.focus();
     } else if (
       lastFocusedElementRef.current &&
       lastFocusedElementRef.current instanceof HTMLElement
     ) {
+      // restore focus, is it necessary?
       lastFocusedElementRef.current.focus();
```

Comment:
Does it have to restore the focus when picker is closed? This seems to be wierd but it may be necessary for accessibility.

Reply:
That's generally a good question to ask since browsers do implement some fixup logic so that you don't have to start from the top if the focused element is removed. But that usually relies on the order in the DOM and therefore breaks with portaled content. It's also just better to be explicit about this so that you don't have to keep in mind where you  cursor was before you opened to *-picker. That would be quite the "stack" you have to keep in your brain.